src/components/NcAppNavigation/NcAppNavigation.vue

```diff
@@ -78,21 +78,29 @@ emit('toggle-navigation', {
 </template>
 
 <script>
-import NcAppNavigationToggle from '../NcAppNavigationToggle/index.js'
 import { useIsMobile } from '../../composables/useIsMobile/index.js'
 import { getTrapStack } from '../../utils/focusTrap.js'
-
 import { emit, subscribe, unsubscribe } from '@nextcloud/event-bus'
-
 import { createFocusTrap } from 'focus-trap'
 
+import NcAppNavigationToggle from '../NcAppNavigationToggle/index.js'
+import Vue from 'vue'
+
 export default {
 	name: 'NcAppNavigation',
 
 	components: {
 		NcAppNavigationToggle,
 	},
 
+	// Injected from NcContent
+	inject: {
+		setHasAppNavigation: {
+			default: () => () => Vue.util.warn('NcAppNavigation is not mounted inside NcContent, this is probably an error.'),
```

Comment:
If we show a warning here, we should add something to the Files app to not show the warning

Reply:
I would say it depends. Is there any good reason to not use `NcContent`?
If yes: Then we can remove the warning and adjust the docs.
If not: We should adjust files to use it.

Reply:
There isn't, this is the only way we ask Devs to build Vue apps